kibana/README.md

```diff
@@ -996,3 +996,64 @@ readonlyrest:
 ```
 
 Now try to login as user1, and ReadonlyREST Enterprise should initialise the index ".kibana\_user1" with all the index patterns and dashboards contained in the template tenancy.
+
+## Impersonation
+
+User with specific roles (`admin` or `unrestricted`) and permissions (`impersonation` section in the `readonlyrest.yml` file)
+can impersonate (imitate behaviors or actions of the user) another user.
+
+### Test settings
+
+During user impersonation, the `reasonlyrest.yml` settings are replaced by test settings and used during a whole impersonation process
+You need to configure test settings to be able to impersonate the user
+
+
+#### Setting test settings
+
+To set test settings:
+
+1. Open the RoR menu
+2. Click the Edit security settings button
+
+![Test settings ror menu](<../.gitbook/assets/test_settings_ror_menu.png>)
+
+3. Go into the Test settings tab
+4. You can set the TTL value. After the TTL value is reached, your test settings will be deactivated automatically
```

Comment:
Acronyms should be expanded and described.

"You can optionally (🙋‍♂️ is it optional? If yes, mention the default behaviour) set the "time to live" (TTL), that is a time interval after which the test settings will be automatically deactivated (🙋‍♂️ will the impersonation session also abruptly exit?)".

Reply:
the same here - maybe we should explain somewhere why the TTL was introduced.

Reply:
`is it optional? If yes, mention the default behavior` It was optional, but based on recent changes, it's required now,

<img width="838" alt="image" src="https://user-images.githubusercontent.com/9945753/168877715-bbcfb897-8a8b-47ef-abe8-651da9d965c0.png">


`will the impersonation session also abruptly exit?`

That's a very good question, Looks like we need to fix it because in the current approach we log out a user (btw we need to fix it because looks like, for some reason, /session-probe endpoint return 200 status, and we probably don't send `x-ror-impersonating` header in this case)

Reply:
I don't remember how you implemented this, and I'm from my phone now. But session probe calls to pkp, then pkp goes to es to refresh the session iirc. So the impersonation header is ok if it's handled by the esclient as part of the authentication headers stored in the session. Right?

Reply:
I changed this logic recently, so

1. /session-probe calls to pkp
2. If "impersonation" mode we ask elasticsearch if the test settings ttl is still valid, and if not, we finish the impersonation session and redirect a user into the impersonation tab

All changed session-probe logic is here https://github.com/sscarduzio/readonlyrest_kbn/pull/303/files#diff-224f72aa407e80ab0821d8da5391a1a0b8ec114175355a10f11048abd749363dR163

Reply:
text changed ✔️

Reply:
> the same here - maybe we should explain somewhere why the TTL was introduced.

@coutoPL And what is the reason to introduce TTL? :P If I understand correctly the reason to introduce it is an optimization of resources, if several users will have test engines loaded, it will block resources somehow (?)

Reply:
The resources are not so important, but it's true. We just want the admin to invalidate the Test Settings after his work. It's just a helper feature for him (aka playground where he is allowed to experiment without breaking the production cluster). For security reasons we also don't want to have the playground active for a long time (because he could forget about the invalidation of the Test Settings)

Reply:
Ok, so the value for the admin is a security, so maybe we can add something like:

 ` "Time to live" (TTL) is introduced to automatically deactivate test settings when your work is finished. For security reasons, it's always good to deactivate your testing playground, and "time to live" (TTL) will ease this process. `

Maybe it's good to write what exactly the security reasons are? (I'm not sure)